generators/chipyard/src/main/scala/Subsystem.scala

```diff
@@ -53,19 +45,7 @@ trait HasChipyardTiles extends HasTiles
   // There is something weird with registering tile-local interrupt controllers to the CLINT.
   // TODO: investigate why
   val tiles = allTilesInfo.sortWith(_._1.hartId < _._1.hartId).map {
-    case (param, crossing) => {
-
-      val tile = param match {
-        case r: RocketTileParams => {
-          LazyModule(new RocketTile(r, crossing, PriorityMuxHartIdFromSeq(rocketTileParams), logicalTreeNode))
-        }
-        case b: BoomTileParams => {
-          LazyModule(new BoomTile(b, crossing, PriorityMuxHartIdFromSeq(boomTileParams), logicalTreeNode))
-        }
-        case a: ArianeTileParams => {
-          LazyModule(new ArianeTile(a, crossing, PriorityMuxHartIdFromSeq(arianeTileParams), logicalTreeNode))
-        }
-      }
+    case (param, crossing, tile) => {
```

Comment:
Note that the sorting by hartid here was to guarantee that the Tiles  were constructed by order of hartid.
Now that you've hoisted the Tile construction out of this block, this is no longer true.

You should modify the definition of allTilesInfo to instantiateTiles by hartid.

Reply:
Also, unrelated, but adding a `require` that no two Tiles have the same hartid would be a good sanity check to perform here.

Reply:
According to the comment, it seems like what matters is the order they are connected to the interrupt controller (which is done by `connectInterrupts()` in the remaining section in this file), so I think it is necessary to be there.

Reply:
I forgot to included the sanity check in the commit I pushed, but I will put it into the code and push it in the next revision.

Reply:
Can you modify core construction to be in order of hartid anyways? The test case which exposed this issue is not easy to recreate, and is not part of the regressions. Minimizing changes to functionality here would be ideal.

Reply:
Sure. I will make another commit later.

Reply:
Done.